### DISCUSSION TITLE
1.0 Release - Info

### DISCUSSION THREAD

**Discussion by SimonMolinsky** [DataverseLabs/pyinterpolate#416] [Announcements]
`Pyinterpolate` is under heavy optimization and refactoring. I plan the 1.0 release in the upcoming year, so stay tuned! If you have any ideas or comments related to the package, feel free to share them, and I will consider every single one!

To let you know, I'm using this package in a business setting on cloud platforms, and it drives some decisions. `Pyinterpolate` will be relatively lightweight. I don't plan to use `scikit-learn` or any package of this size as a dependency. The package will be modular, and some modules and dependencies will be optional.

**Comment by leouieda** [comment]
@SimonMolinsky interested to see what changes in the new version! 

If you can share, I'd love to know what about the cloud deployment prompted the decision to keep this lightweight. Scikit-learn is a heavy dependency but tends to be well supported. So I was curious (since I depend on it myself in other projects).

**Comment by SimonMolinsky** [comment]
@leouieda `scikit-learn` gets great support from every cloud service when you want to deploy ML pipelines, and it is the best package for data scientists in my opinion. But on multiple occasions, I have had to deploy a small Lambda or GC Function, which does only one thing. For example - it performs automatic Ordinary Kriging interpolation as an intermediate step between the BigQuery table and the client-server. Using multiple Python packages in a single environment slows down lambdas; if packages are too big, I must push them inside containers. And it means that warming-up time is even longer and I pay more for the code that does nothing. `scikit-learn` is one example, but another could be `matplotlib`. I don't want to resign from plotting capabilities because Kriging without supervision is a risky business, but I will move all visualizations into a dedicated module which could be excluded in the installation for the cloud or data processing environment.

For me the biggest pain is the universal kriging implementation. Without regression from `scikit-learn` or `statsmodels` it seems to be a cumbersome task. Probably it can be done within additional submodule, so finally `sklearn` will be imported somewhere in the package :)
